ocamltest/filecompare.ml

```diff
@@ -104,7 +105,28 @@ let line_seq_of_in_channel ~normalise ic =
     in
       read_line
 
-let compare_text_files ignored_lines file1 file2 =
+(* [seq_equal_ignore_last2 n seq1 seq2] returns [true] if and only if [seq2],
+   after removing the [n] last elements, equals [seq1].  If [seq2] has
+   less than [n] elements, then it returns [true] if and only if [seq1] is
+   empty. *)
+let seq_equal_ignore_last2 n seq1 seq2 =
+  (* If [seq1] is empty, just check the length of [seq2]. Otherwise, check that
+     [seq1] is a prefix of [seq2] with exactly [n] elements remaining. *)
+  let open Seq in
+  match seq1 () with
+  | Nil -> length seq2 <= n
+  | Cons _ as node1 ->
+      let seq1 = fun () -> node1 in
+      let rec loop seq1 seq2 =
+        match seq1 (), seq2 () with
+        | Nil, node2 -> length (fun () -> node2) = n
+        | Cons _, Nil -> false
+        | Cons (x1, rem1), Cons (x2, rem2) ->
+            if x1 = x2 then loop rem1 rem2 else false
+      in
+      loop seq1 seq2
+
```

Comment:
That's a lot of code...  I'm afraid we need a review by @shindere or someone who is familiar with ocamltest.